File: MobileMeteoTests/MMTMeteorogramQueryTests.m
//
//  MMTMeteorogramQueryTests.m
//  MobileMeteo
//
//  Created by Kamil on 05.06.2015.
//  Copyright (c) 2015 Kamil Szostakowski. All rights reserved.
//

#import <UIKit/UIKit.h>
#import <XCTest/XCTest.h>
#import <CoreLocation/CoreLocation.h>

#import "MMTMeteorogramQuery.h"

@interface MMTMeteorogramQueryTests : XCTestCase

@property (nonatomic, strong) CLLocation *location;
@property (nonatomic, strong) NSDateFormatter *formatter;
@end

@implementation MMTMeteorogramQueryTests

- (void)setUp
{
    [super setUp];
    
    self.location = [[CLLocation alloc] initWithLatitude:53.03 longitude:18.57];
    self.formatter = [NSDateFormatter new];
    self.formatter.dateFormat = @"YYYY-MM-dd'T'HH:mm";
    self.formatter.timeZone = [NSTimeZone timeZoneForSecondsFromGMT:7200];
}

- (void)tearDown
{
    self.formatter = nil;
    
    [super tearDown];
}

- (void)testEmptyInitialization
{
    MMTMeteorogramQuery *query = [[MMTMeteorogramQuery alloc] initWithLocation:self.location date:nil];
    XCTAssertNil(query);
}

- (void)testCoordinate
{
    NSDate *date = [self.formatter dateFromString:@"2015-04-15T10:00"];
    MMTMeteorogramQuery *query = [[MMTMeteorogramQuery alloc] initWithLocation:self.location date:date];
    
    XCTAssertEqual(53.03, query.location.coordinate.latitude);
    XCTAssertEqual(18.57, query.location.coordinate.longitude);
}

- (void)testDateBeforeNoon
{
    NSDate *date = [self.formatter dateFromString:@"2015-04-15T10:00"];
    MMTMeteorogramQuery *query = [[MMTMeteorogramQuery alloc] initWithLocation:self.location date:date];
    
    XCTAssertEqualObjects(@"2015041500", query.date);
}

- (void)testDateAfterNoonOfLocalTimeZone
{
    NSDate *date = [self.formatter dateFromString:@"2015-02-13T13:00"];
    MMTMeteorogramQuery *query = [[MMTMeteorogramQuery alloc] initWithLocation:self.location date:date];
    
    XCTAssertEqualObjects(@"2015021300", query.date);
}

- (void)testDateAfterNoonOfGMTTimeZone
{
    NSDate *date = [self.formatter dateFromString:@"2015-02-13T14:00"];
    MMTMeteorogramQuery *query = [[MMTMeteorogramQuery alloc] initWithLocation:self.location date:date];
    
    XCTAssertEqualObjects(@"2015021312", query.date);
}

@end

File: MobileMeteoTests/MMTMeteorogramQueryTests.swift
//
//  MMTMeteorogramQueryTests.swift
//  MobileMeteo
//
//  Created by Kamil Szostakowski on 30.06.2015.
//  Copyright (c) 2015 Kamil Szostakowski. All rights reserved.
//

import XCTest
import Foundation
import CoreLocation
import MobileMeteo

class MMTMeteorogramQueryTests : XCTestCase
{
    // MARK: Setup
    
    var location : CLLocation!
    var formatter : NSDateFormatter!
    
    override func setUp()
    {
        super.setUp()
        
        location = CLLocation(latitude: 53.03, longitude: 18.57)
        formatter = NSDateFormatter()
        formatter.dateFormat = "YYYY-MM-dd'T'HH:mm"
        formatter.timeZone = NSTimeZone(forSecondsFromGMT:7200)
    }
    
    override func tearDown()
    {
        location = nil;
        formatter = nil;
        
        super.tearDown()
    }
    
    // MARK: Test methods    
    
    func testDate()
    {
        let date = formatter.dateFromString("2015-04-15T10:00")!
        let query = MMTMeteorogramQuery(location:location, date: date, name: nil)
        
        XCTAssertEqual(date, query.date)
    }
    
    func testCoordinate()
    {
        let query = MMTMeteorogramQuery(location:location)
    
        XCTAssertEqual(53.03, query.location.coordinate.latitude)
        XCTAssertEqual(18.57, query.location.coordinate.longitude)
    }
    
    func testName()
    {
        let query = MMTMeteorogramQuery(location: location, name: "Example")
        
        XCTAssertEqual("Example", query.locationName!)
    }        
}